### ISSUE TITLE
Crash? No tray icon

### ISSUE THREAD

**Issue by ghost** [buzz/volctl#20]
Hi there, trying to use this on Arch with xfce. When starting volctl I get the following which just keeps repeating. There is no icon/applet in my system tray. Not sure if this is an error?

``` 
~ 
 02:15:55 AM   >> $ volctl
PulseAudio: Connection ready
Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 232, in 'calling callback function'
  File "/usr/lib/python3.7/site-packages/volctl/pulseaudio.py", line 203, in _pa_sink_info_cb
    self._dict_from_proplist(struct.contents.proplist))
  File "/usr/lib/python3.7/site-packages/volctl/pulseaudio.py", line 312, in _on_new_pa_sink
    self._pa_sinks[index].update(struct)
  File "/usr/lib/python3.7/site-packages/volctl/pulseaudio.py", line 362, in update
    if self == self.pa_mgr.get_first_sink():
  File "/usr/lib/python3.7/site-packages/volctl/pulseaudio.py", line 269, in get_first_sink
    return self._pa_sinks[0]
KeyError: 0
```
Hitting ctrl+c also gives:

```
^CTraceback (most recent call last):
  File "/usr/bin/volctl", line 11, in <module>
    load_entry_point('volctl==0.6.1', 'gui_scripts', 'volctl')()
  File "/usr/lib/python3.7/site-packages/volctl/__main__.py", line 11, in main
    Gtk.main()
  File "/usr/lib/python3.7/site-packages/gi/overrides/Gtk.py", line 1641, in main
    return _Gtk_main(*args, **kwargs)
  File "/usr/lib/python3.7/contextlib.py", line 119, in __exit__
    next(self.gen)
  File "/usr/lib/python3.7/site-packages/gi/_ossighelper.py", line 251, in register_sigint_fallback
    signal.default_int_handler(signal.SIGINT, None)
KeyboardInterrupt
```
I initially installed via pakku but using makepkg results in the same.
Probably prudent to link to the AUR package.
https://aur.archlinux.org/packages/volctl/
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=volctl

I also tried manually doing
`sudo glib-compile-schemas /usr/share/glib-2.0/schemas/`
since it was missing from the PKGBUILD but that results in the same.

**Comment by buzz** [comment]
There is a bug in `volctl/pulseaudio.py` that leads to an unhandled KeyError.

The real question is why there's no first audio sink on your device which should be the master volume.

Could you try the fix in branch `2-fix-keyerror`?

**Comment by HellishINC** [comment]
To be completely honest,I don't understand much of how pulse works. Could it be that the reason there's no first sink is because I have it to off?

I use two audio devices on my machine. The "primary" is HDMI audio via my GTX970. Since that doesn't support any other out/inputs I also have a dedicated Xonar D1 which handles my headphones. I also have an USB microphone for input (which is also has "stereo output"? don't ask me).

I'll give that branch a shot either today or tomorrow. Been really hetic as of late. Thanks for the respone and happy holidays to you!

**Comment by buzz** [comment]
> To be completely honest, I don't understand much of how pulse works. Could it be that the reason there's no first sink is because I have it to off?

Yes and no. This is really a bug in volctl. I figured PulseAudio actually returns a "DummyOutput" device if I manually disable all my audio interfaces. The problem occurs because the index seems to shift when I re-enable the interfaces and the code was expecting the first audio interface always show as index `0`.

> I'll give that branch a shot either today or tomorrow. Been really hetic as of late. Thanks for the response and happy holidays to you!

Yep, no hurry. I updated the branch again. It does work now on my end. Still I would be very interested if it works for you.

Same to you!
